patch/ros-humble-examples-rclcpp-minimal-subscriber.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 617c70014..049458fcc 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -14,27 +14,28 @@ find_package(ament_cmake REQUIRED)
+ find_package(rclcpp REQUIRED)
+ find_package(rclcpp_components REQUIRED)
+ find_package(std_msgs REQUIRED)
++find_package(Python COMPONENTS Interpreter Development REQUIRED)
+ 
+ add_executable(subscriber_lambda lambda.cpp)
+-ament_target_dependencies(subscriber_lambda rclcpp std_msgs)
++target_link_libraries(subscriber_lambda PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_member_function member_function.cpp)
+-ament_target_dependencies(subscriber_member_function rclcpp std_msgs)
++target_link_libraries(subscriber_member_function PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_member_function_with_topic_statistics member_function_with_topic_statistics.cpp)
+-ament_target_dependencies(subscriber_member_function_with_topic_statistics rclcpp std_msgs)
++target_link_libraries(subscriber_member_function_with_topic_statistics PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_member_function_with_type_adapter member_function_with_type_adapter.cpp)
+-ament_target_dependencies(subscriber_member_function_with_type_adapter rclcpp std_msgs)
++target_link_libraries(subscriber_member_function_with_type_adapter PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_member_function_with_unique_network_flow_endpoints member_function_with_unique_network_flow_endpoints.cpp)
+-ament_target_dependencies(subscriber_member_function_with_unique_network_flow_endpoints rclcpp std_msgs)
++target_link_libraries(subscriber_member_function_with_unique_network_flow_endpoints PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_not_composable not_composable.cpp)
+-ament_target_dependencies(subscriber_not_composable rclcpp std_msgs)
++target_link_libraries(subscriber_not_composable PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_executable(subscriber_content_filtering content_filtering.cpp)
+-ament_target_dependencies(subscriber_content_filtering rclcpp std_msgs)
++target_link_libraries(subscriber_content_filtering PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
+ 
+ add_library(wait_set_subscriber_library SHARED
+     wait_set_subscriber.cpp
```

Comment:
Can you explain why C++ examples need to link Python? To avoid unintended side-effects (especially in macos) could it make sense to make these patches emscripten specific?

Reply:
Agreed - we can simply rename the relevant patches to `.emscripten.patch`